runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -7129,6 +7129,194 @@ static TR::Register *inlineStringLatin1Inflate(TR::Node *node, TR::CodeGenerator
    return NULL;
    }
 
+/**
+ * \brief
+ *   Generate inlined instructions equivalent to java/lang/StringCoding.hasNegatives or java/lang/StringCoding.countPositives
+ *
+ * \param node
+ *   The tree node
+ *
+ * \param recognizedMethod
+ *   The method being inlined, should be either hasNegatives or countPositives
+ *
+ * \param cg
+ *   The Code Generator
+ *
+ * Note that this version does not support discontiguous arrays
+ */
+static TR::Register *inlineHasNegativesOrCountPositives(TR::Node *node, bool isHasNegatives, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT_FATAL(!TR::Compiler->om.canGenerateArraylets(), "Discontiguous array is not supported");
+
+   // Registers
+   TR::Register *arrayReg = cg->evaluate(node->getChild(0));
+   TR::Register *offsetReg = cg->evaluate(node->getChild(1));
+   TR::Register *lengthReg = cg->evaluate(node->getChild(2));
+
+   TR::Register *dataAddrReg = cg->allocateRegister(TR_GPR);
+   TR::Register *countReg = cg->allocateRegister(TR_GPR);
+   TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
+   TR::Register *vtmpReg = cg->allocateRegister(TR_VRF);
+   TR::Register *indexReg;
+   if (node->getChild(1)->getReferenceCount() == 1)
+      {
+      indexReg = offsetReg;
+      }
+   else
+      {
+      indexReg = cg->allocateRegister(TR_GPR);
+      generateMovInstruction(cg, node, indexReg, offsetReg);
+      }
+
+   TR::RegisterDependencyConditions *dependencies = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(7, 7, cg->trMemory());
```

Comment:
I think `offsetReg` needs to be part of this (so 8 dependencies), unless I am overlooking something. It is used in the internal control flow code below.

Reply:
Correct.  `offsetReg` needs to be preserved.
I updated the code.